### PR TITLE
Improve support for distributed guest authors

### PR DESCRIPTION
**Pull request by leogermani**
Handle guest authors distributed via Newspack Network and Distributor plugins

### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Improves the support for the distributed post authors by Newspack Network. 

Basically - remove the `a` tag for authors without a URL.

Also sends an extra information in the `newspack_author_bio_name` filter.

### How to test the changes in this Pull Request:

See https://github.com/Automattic/newspack-network/pull/25

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by miguelpeixe** [comment]
Can't this be a filter inside the Newspack Network plugin so it's contained in the same codebase? Distributor has a similar strategy to filter the author to make it appear as the site name.

**Comment by miguelpeixe** [comment]
Thank you for explaining!

**Review by laurelfulford — CHANGES_REQUESTED** [review]
This is so cool, @leogermani! 🙌 

I tested posts with single guest authors, multiple guest authors, mix of guest and regular, and a guest mapped to a regular author. The only issue I've run into is that the distributed guest author avatars don't seem to be syncing, but only for the byline at the top -- they appear as expected in the author bio at the bottom of the post. The regular author avatars work 100% as expected. 

Screen Shot 2023-12-21 at 10 29 04 AM

Screen Shot 2023-12-21 at 10 29 10 AM

Job titles, bios, and email addresses were brought over as expected!

**Styles**
There was some discussion about whether these should be styled to match the linked authors. Personally, I'm leaning towards leaving them as the default grey -- I think that could give some visual feedback that they're not the same as the other names and (hopefully!) help set expectations that they're not going to be links. I'm open to discussing that further, though! I feel like either way we land with this, there will be publishers who want it to work "the other way" (either match the regular links, or look less like the links).

**Custom CSS**
I think there's a _small_ risk existing custom CSS won't be applied as expected to these new unlinked bylines because an HTML element has been removed. I think it's super-minor though -- and it won't affect existing posts since it's a new feature, just posts going forward. I think we can tackle this as it comes up (it will likely also happen if we opt to style these to match the links).

**Review by laurelfulford — CHANGES_REQUESTED** [review]
Thanks @leogermani! [8367482](https://github.com/Automattic/newspack-theme/pull/2215/commits/836748285ce6b0e10b3f32389592a0dd8f780d06) fixed the issue for me:

image

I have one more change that I missed in my initial review: Newspack Sacha also has its own [/author-bio.php file](https://github.com/Automattic/newspack-theme/blob/master/newspack-sacha/template-parts/post/author-bio.php) that would need the same changes as the main theme's file, just to make sure it's picking up the same updated author info. The separate file was to help make its centred layout easier to accomplish, but more than once I wish I'd just reworked the original file earlier on to avoid adding it!

**Comment by leogermani** [comment]
Thanks for the review @laurelfulford 

I've updated the checks with some recent changes on the PR in the Newspack Network plugin, but it should work just the same.

And I applied the changes to the Sacha theme.

All those `if`s are not pretty, but I think it's good enough for now.

I also tested with co-authors-plus disabled in both themes just to make sure :)

**Review by laurelfulford — APPROVED** [review]
Thanks @leogermani! 

> All those ifs are not pretty, but I think it's good enough for now.

I thought you meant the ones that were already there 😆 These files are already full of so many `if`s and `else`s, adding more at this point can't hurt!

I retested this with Newspack and Newspack Sasha, with and without Co-Author Plus, and the single posts work as expected! 🎉 

I found one more spot that may need changes, but it's not something that's actually being addressed in this PR yet, so I'm marking this as approved in case it's just easier to do it as a separate fix: clicking around, I noticed that the 'real' authors that are brought over via the Network still have the site's name as their name on the author archives. For example, this author has the display name 'laurel' but it's still showing 'Newspack Dev' (my spoke site) on the Hub archive: 

image

Just let me know if you have any questions about that, or can't reproduce! I don't remember spotting that when I last tested, but I don't think I actually clicked over to the transferred author archive pages. (Also just let me know if I should spin up an Asana task for that with better steps to replicate!).

**Comment by leogermani** [comment]
@laurelfulford Can you check if you unmarked the "Override Author Byline" option in Distributor > Settings?

image

**Comment by laurelfulford** [comment]
> Can you check if you unmarked the "Override Author Byline" option in Distributor > Settings?

@leogermani I had it checked when this happened, but I also unchecked it both on the Hub and Spoke and could see the same issue 😕 I'll take some time this AM trying to recreate in a different set up, and can follow up with a separate task if I'm still seeing it with a link!

**Comment by laurelfulford** [comment]
@leogermani Just wanted to follow up and confirm I CAN'T recreate the above on two JN sites set up with Hub & Spoke plus this PR, so that issue was totally user error 🙂
